crates/bitcoin-da/src/helpers/parsers.rs

```diff
@@ -200,338 +509,219 @@ pub fn parse_hex_transaction(
-
-        assert!(result.is_err(), "Failed to error on no publickey tag.");
-        assert_eq!(result.unwrap_err(), ParserError::EnvelopeHasIncorrectFormat);
-
-        // body
-        let reveal_script_builder = script::Builder::new()
-            .push_x_only_key(&XOnlyPublicKey::from_slice(&[1; 32]).unwrap())
-            .push_opcode(OP_CHECKSIG)
-            .push_opcode(OP_FALSE)
-            .push_opcode(OP_IF)
-            .push_slice(PushBytesBuf::try_from(ROLLUP_NAME_TAG.to_vec()).unwrap())
-            .push_slice(PushBytesBuf::try_from("sov-btc".as_bytes().to_vec()).unwrap())
-            .push_slice(PushBytesBuf::try_from(SIGNATURE_TAG.to_vec()).unwrap())
-            .push_slice(PushBytesBuf::try_from(vec![0u8; 64]).unwrap())
-            .push_slice(PushBytesBuf::try_from(PUBLICKEY_TAG.to_vec()).unwrap())
-            .push_slice(PushBytesBuf::try_from(vec![0u8; 64]).unwrap())
-            .push_slice(PushBytesBuf::try_from(RANDOM_TAG.to_vec()).unwrap())
-            .push_int(0)
-            .push_opcode(OP_ENDIF);
-
-        let reveal_script = reveal_script_builder.into_script();
-
-        let result =
-            parse_relevant_inscriptions(&mut reveal_script.instructions().peekable(), "sov-btc");
-
-        assert!(result.is_err(), "Failed to error on no body tag.");
+    fn only_checksig() {
+        let header = TransactionHeaderLightClient {
+            rollup_name: b"sov-btc",
+            kind: TransactionKindLightClient::Complete,
+        };
 
-        // random
         let reveal_script_builder = script::Builder::new()
             .push_x_only_key(&XOnlyPublicKey::from_slice(&[1; 32]).unwrap())
-            .push_opcode(OP_CHECKSIG)
-            .push_opcode(OP_FALSE)
-            .push_opcode(OP_IF)
-            .push_slice(PushBytesBuf::try_from(ROLLUP_NAME_TAG.to_vec()).unwrap())
-            .push_slice(PushBytesBuf::try_from("sov-btc".as_bytes().to_vec()).unwrap())
-            .push_slice(PushBytesBuf::try_from(SIGNATURE_TAG.to_vec()).unwrap())
-            .push_slice(PushBytesBuf::try_from(vec![0u8; 64]).unwrap())
-            .push_slice(PushBytesBuf::try_from(PUBLICKEY_TAG.to_vec()).unwrap())
-            .push_slice(PushBytesBuf::try_from(vec![0u8; 64]).unwrap())
-            .push_slice(PushBytesBuf::try_from(BODY_TAG.to_vec()).unwrap())
-            .push_slice(PushBytesBuf::try_from(vec![0u8; 128]).unwrap())
-            .push_opcode(OP_ENDIF);
+            .push_opcode(OP_CHECKSIGVERIFY)
+            .push_slice(PushBytesBuf::try_from(header.to_bytes()).expect("Cannot push header"));
 
         let reveal_script = reveal_script_builder.into_script();
 
-        let result =
-            parse_relevant_inscriptions(&mut reveal_script.instructions().peekable(), "sov-btc");
-
-        assert!(result.is_err(), "Failed to error on no random tag.");
-        assert_eq!(result.unwrap_err(), ParserError::EnvelopeHasIncorrectFormat);
-    }
-
-    #[test]
-    fn non_parseable_tx() {
```

Comment:
this one too

Reply:
What's with this one?

Reply:
i guess some of my comments are missing...

non_parseable_tx + leave_out_tags

we still need these tests.

Reply:
non_parseable_tx - for this one we need to create a tx with a correct kind/rollup name but with a mess inside the script. is it useful?

leave_out_tags - since our tx format changed for a better parsing there is no need for a state machine to match tags as before. Signature, pub key, nonce - everything has its own place in the script instead of [tag] -> [data].